lib/engine/game/g_1847_ae/game.rb

```diff
@@ -111,6 +123,57 @@ def operating_round(round_num)
             [Engine::Step::BuyCompany, { blocks: true }],
           ], round_num: round_num)
         end
+
+        def saar
+          corporation_by_id('Saar')
+        end
+
+        def hlb
+          corporation_by_id('HLB')
+        end
+
+        def init_share_pool
+          G1847AE::SharePool.new(self)
+        end
+
+        # Reserved share type used to represent investor shares
+        def ipo_reserved_name(_entity = nil)
+          'Investor'
+        end
+
+        def init_corporations(stock_market)
+          self.class::CORPORATIONS.map do |corporation|
+            par_price = stock_market.par_prices.find { |p| p.price == corporation[:required_par_price] }
+            corporation[:par_price] = par_price
+            G1847AE::Corporation.new(
+              min_price: par_price.price,
+              capitalization: self.class::CAPITALIZATION,
+              **corporation.merge(corporation_opts),
+            )
+          end
+        end
+
+        def setup
+          # Reserve investor shares and add money for them to treasury
+          [saar.shares[1], saar.shares[2], hlb.shares[1]].each { |s| s.buyable = false }
+          saar.cash += saar.par_price.price * 2
+          hlb.cash += hlb.par_price.price
+        end
```

Comment:
do saar and hlb have a par price at the start of the game? shouldn't this code run after parring?

Reply:
They do, in this game you don't choose pars.